src/FMBot.Images/Generators/PuppeteerService.cs

```diff
@@ -719,12 +720,13 @@ public void CreatePopularityIcebergImage(SKBitmap chartImage, string username, s
                     a.Popularity > minPopularity && a.Popularity <= maxPopularity)
                 .OrderByDescending(a => a.Playcount)
                 .Take(5)
+                .OrderByDescending(a => a.Name.Length)
                 .ToList();
 
             var sectionY = FirstArtistY + (section * (totalAvailableHeight / TotalSections));
             var positions = GenerateArtistNamePositions(sectionY, sectionArtists.Count);
 
-            for (var i = 0; i < Math.Min(sectionArtists.Count, 5); i++)
+            for (var i = 0; i < sectionArtists.Count; i++)
```

Comment:
unnecessary change perhaps, i can amend it and force push again if you'd like. i just think the `.Min` call is not needed when `.Take` takes up to 5 elements already.